system/include/SSE/emmintrin.h

```diff
@@ -0,0 +1,1621 @@
+{
+  return _mm_move_sd(__a, _mm_cmpneq_pd(__a, __b));
+}
+
+static __inline__ __m128d __attribute__((__always_inline__, __nodebug__))
+_mm_cmpnlt_sd(__m128d __a, __m128d __b)
+{
+  return _mm_move_sd(__a, _mm_cmpnlt_pd(__a, __b));
+}
+
+static __inline__ __m128d __attribute__((__always_inline__, __nodebug__))
+_mm_cmpnle_sd(__m128d __a, __m128d __b)
+{
+  return _mm_move_sd(__a, _mm_cmpnle_pd(__a, __b));
+}
+
+static __inline__ __m128d __attribute__((__always_inline__, __nodebug__))
+_mm_cmpngt_sd(__m128d __a, __m128d __b)
+{
+  return _mm_move_sd(__a, _mm_cmpngt_pd(__a, __b));
+}
+
+static __inline__ __m128d __attribute__((__always_inline__, __nodebug__))
+_mm_cmpnge_sd(__m128d __a, __m128d __b)
+{
+  return _mm_move_sd(__a, _mm_cmpnge_pd(__a, __b));
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_comieq_sd(__m128d __a, __m128d __b)
+{
+  return ((__f64x2)__a)[0] == ((__f64x2)__b)[0];
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_comilt_sd(__m128d __a, __m128d __b)
+{
+  return ((__f64x2)__a)[0] < ((__f64x2)__b)[0];
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_comile_sd(__m128d __a, __m128d __b)
+{
+  return ((__f64x2)__a)[0] <= ((__f64x2)__b)[0];
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_comigt_sd(__m128d __a, __m128d __b)
+{
+  return ((__f64x2)__a)[0] > ((__f64x2)__b)[0];
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_comige_sd(__m128d __a, __m128d __b)
+{
+  return ((__f64x2)__a)[0] >= ((__f64x2)__b)[0];
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_comineq_sd(__m128d __a, __m128d __b)
+{
+  return ((__f64x2)__a)[0] != ((__f64x2)__b)[0];
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_ucomieq_sd(__m128d __a, __m128d __b)
+{
+  return ((__f64x2)__a)[0] == ((__f64x2)__b)[0];
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_ucomilt_sd(__m128d __a, __m128d __b)
+{
+  return ((__f64x2)__a)[0] < ((__f64x2)__b)[0];
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_ucomile_sd(__m128d __a, __m128d __b)
+{
+  return ((__f64x2)__a)[0] <= ((__f64x2)__b)[0];
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_ucomigt_sd(__m128d __a, __m128d __b)
+{
+  return ((__f64x2)__a)[0] > ((__f64x2)__b)[0];
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_ucomige_sd(__m128d __a, __m128d __b)
+{
+  return ((__f64x2)__a)[0] >= ((__f64x2)__b)[0];
+}
+
+static __inline__ int __attribute__((__always_inline__, __nodebug__))
+_mm_ucomineq_sd(__m128d __a, __m128d __b)
+{
+  return ((__f64x2)__a)[0] != ((__f64x2)__b)[0];
+}
+
+static __inline__ __m128 __attribute__((__always_inline__, __nodebug__))
+_mm_cvtpd_ps(__m128d __a)
+{
+  return (__m128) { (float)__a[0], (float)__a[1], 0, 0 };
```

Comment:
wasm_simd128.h provides a `wasm_f32x4_make` convenience function that would be good to use here, again to avoid using the vector extensions directly.

Reply:
changed to `wasm_f32x4_make`